frontend/app/work_packages/controllers/work-package-new-controller.js

```diff
@@ -0,0 +1,172 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+module.exports = function(
+           $scope,
+           $rootScope,
+           $state,
+           $stateParams,
+           $timeout,
+           $window,
+           PathHelper,
+           WorkPackagesOverviewService,
+           WorkPackageFieldService,
+           WorkPackageService,
+           EditableFieldsState
+           ) {
+
+  var vm = this,
+      unhideableFields = [
+        'subject',
+        'type',
+        'status',
+        'description',
+        'priority',
+        'assignee',
+        'percentageDone'
+      ];
+
+  vm.groupedFields = [];
+  vm.hideEmptyFields = true;
+
+  vm.submit = submit;
+  vm.cancel = cancel;
+
+  vm.loaderPromise = null;
+
+  vm.isGroupHideable = isGroupHideable;
+  vm.isFieldHideable = isFieldHideable;
+  vm.getLabel = getLabel;
+  vm.isSpecified = isSpecified;
+  vm.isEditable = isEditable;
+  vm.hasNiceStar = hasNiceStar;
+  vm.showToggleButton = showToggleButton;
+
+  activate();
+
+  function activate() {
+    EditableFieldsState.forcedEditState = true;
+    vm.loaderPromise = WorkPackageService.initializeWorkPackage($scope.projectIdentifier, {
+      type: PathHelper.apiV3TypePath($stateParams.type)
```

Comment:
Disclaimer: I don't know the surrounding/calling code. From a wishful unicorn perspective I would love the following:

* `$stateParams.type` is a type resource as returned by APIv3
* you don't need to use a pathhelper, but simply use `type.links.self.href`

If you say that would require lots of rewriting we should do it later (hope that does not mean "never"). But judging from other code in the frontend, chances are high, that the source of `$stateParams.type` was already a "good"/"full" type and only its id got ripped off. In that case the change would be easy and I'd request it.